sphinx/themes/basic/static/searchtools.js

```diff
@@ -47,6 +47,14 @@ if (typeof Scorer === "undefined") {
   };
 }
 
+// Global search result kind enum, used by themes to style search results.
+class SearchResultContext {
```

Comment:
```suggestion
class SearchResultContext {
```

Maybe a nitpick, but: this was a `const` previously, compared to a `class` now - I'm not familiar enough with JavaScript to know the implications - are there any drawbacks to this change?

Reply:
This was a change made by AA-Turner (originally in ebbcb364, which I've squashed). I can't tell about the advantages or disadvantages, but it was intentional, so I assume it is the way to go.

Reply:
Ok, thanks (I was too focused on the naming discussion to notice that, evidently :)).

I'll do some background reading out of curiosity (I do notice that the `class` keyword doesn't appear elsewhere in the `basic` theme JavaScript code, and that makes me a little cautious).

Reply:
Well, it was introduced in ES6 (2015) -- and so was `const` that we already use frequently in the `searchtools.js` code -- so there's no compatibility-level concern.  This seems fine to me - I'll re-scan the rest of the changeset again in a few moments.

(I did notice the word 'kind' remains in a comment -- probably worth updating that to 'context' for consistency?)

Reply:
Perhaps I am overthinking this, but I think that for both `SearchResultContext` and also the already-existing `Scorer` values, we probably really want immutable, enum-like behaviour.

Yes, OK, perhaps some plugins/themes could usefully and benevolently override them from their own JavaScript code -- but I feel that those use cases should be rare.

Again I'm not an expert but from a quick search, I wonder whether `Object.freeze` is worth considering for both of them: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

Reply:
Remark: As long as it does not change behavior, we can always improve the JavaScript later. I wouldn't consider the searchtools.js content as public, but YMMV.

Reply:
``class X {}`` is the same as ``const X = class {}`` -- it is syntactic sugar. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#defining_classes

Reply:
> Remark: As long as it does not change behavior, we can always improve the JavaScript later. I wouldn't consider the searchtools.js content as public, but YMMV.

My understanding of JavaScript is fairly outdated, but I had a sense that any code we add at the top-level pretty much does become public, and that making it private and/or immutable requires special care.

Sufficiently-weird use-cases that rely on released functionality can be difficult to distinguish from denials-of-maintainer-time, so I think we should be careful about the features we do make public.

> `class X {}` is the same as `const X = class {}` -- it is syntactic sugar. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#defining_classes

Yep, true - but the change in https://github.com/sphinx-doc/sphinx/commit/ebbcb364cb63737b92f6b76a2fb35334f1df4709 was from `const X = { ... }` (a constant initialized to an object value) to `class X { ... }`.  I don't yet know whether we want/need `class` semantics -- and am (still) wondering whether `Object.freeze` and perhaps [`Symbol`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) (also ref: [this article](https://hacks.mozilla.org/2015/06/es6-in-depth-symbols/)) could make sense - the goal being to achieve something like an immutable, protected enum.